src/Query/TaskGroupingTree.ts

```diff
@@ -37,34 +37,34 @@ export class IntermediateTaskGroupsStorage extends Map<string[], Task[]> {}
 class TaskGroupingTreeNode extends GroupingTreeNode<Task> {}
 
 /**
- * IntermediateTaskGroups does the initial grouping together of tasks,
+ * TaskGroupingTree does the initial grouping together of tasks,
  * in alphabetical order by group names.
  *
- * It is essentially a thin wrapper around Map - see IntermediateTaskGroupsStorage.
+ * It is essentially a thin wrapper around Map - see {@link TaskGroupingTreeStorage}.
  *
- * It is named "Intermediate" because its results are only temporary.
- * They will be discarded once the final TaskGroups object is created.
+ * It is an implementation detail of the task-grouping code, and does not need to
+ * be understood in order to group tasks.
  *
  * Ideally, this code would be simplified and moved in to TaskGroups.
  */
-export class IntermediateTaskGroups {
-    public groups = new IntermediateTaskGroupsStorage();
+export class TaskGroupingTree {
+    public groups = new TaskGroupingTreeStorage();
 
     /**
      * Group a list of tasks, according to one or more task properties
-     * @param groupings 0 or more Grouping values, one per 'group by' line
+     * @param groupers 0 or more Grouping values, one per 'group by' line
```

Comment:
We got rid of the `grouping` in the code, I think this `0 or more Grouping values` should be something else too.